dashboard/app/models/user_level.rb

```diff
@@ -109,12 +109,14 @@ def locked?(stage)
   end
 
   def self.update_lockable_state(user_id, level_id, script_id, locked, readonly_answers)
-    user_level = UserLevel.find_by(user_id: user_id, level_id: level_id, script_id: script_id)
+    user_level = UserLevel.find_or_initialize_by(
+      user_id: user_id,
+      level_id: level_id,
+      script_id: script_id
+    )
 
     # no need to create a level if it's just going to be locked
-    return if !user_level && locked
-
-    user_level ||= UserLevel.create(user_id: user_id, level_id: level_id, script_id: script_id)
```

Comment:
Ahh, I this is what I originally wanted, but only knew about `find_or_create_by`. If I read this correctly, it seems we should still end up not persisting a new user_level to the db in the case where locked is true (i.e. we will initialize one, but it will never actually be saved)?